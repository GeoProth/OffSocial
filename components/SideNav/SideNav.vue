<template>
  <div class="side-nav-container">
    <div class="dashboard" @click="route('/dashboard')">
      <h3>Dashboard</h3>
    </div>
    <div class="privacy" @click="route('/privacy')">
      <h3>Privacy Policy</h3>
    </div>
    <div class="terms" @click="route('/terms')">
      <h3>Terms</h3>
    </div>
  </div>
</template>
<script>
export default {
  name: "SideNav",
  methods: {
    route(link) {
      this.$store.dispatch("layout/toggleSideNav");
      this.$router.push(link);
    }
  }
};
</script>
<style lang="stylus" scoped>
.side-nav-container {
  width: 12rem;
  background-color: darken(white, 20%);

  div {
    height: 3rem;
    padding: 2.5px;
    border: 1px solid lighten(ui_blue, 20%);
    background-color: lighten(ui_blue, 10%);
    color: ui_offwhite;
    display: flex;
    justify-content: center;
    flex-direction: column;
    cursor: pointer;
    &:hover {
      background-color: ui_blue;
    }
  }
  a {
    text-decoration: none;
    color: ui_white;
    vertical-align: -50%;
  }
  h3 {
    margin: 0
    text-align: center
    font-size: 1rem

  }
}
.side-nav-container div {
  width: 100%
}
</style>
