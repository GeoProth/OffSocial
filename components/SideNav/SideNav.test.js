import SideNav from "./SideNav";
describe("Side nav menu", () => {
  let wrapper;
  const mockToggle = jest.fn();
  beforeEach(() => {
    mockToggle.mockReset();
    wrapper = shallowMock(SideNav, {
      store: {
        layout: {
          actions: {
            toggleSideNav: mockToggle
          }
        }
      }
    });
  });
  test("can render", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("clicking a link in the SideNav will route to the new page and then close the SideNav", () => {
    wrapper.find(".dashboard").trigger("click");
    expect(mockToggle).toBeCalled();
    expect(routerPush).toBeCalledWith("/dashboard");
  });
});
