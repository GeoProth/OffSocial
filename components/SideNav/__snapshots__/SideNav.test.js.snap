// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side nav menu can render 1`] = `
<div class="side-nav-container">
  <div class="dashboard">
    <h3>Dashboard</h3>
  </div>
  <div class="privacy">
    <h3>Privacy Policy</h3>
  </div>
  <div class="terms">
    <h3>Terms</h3>
  </div>
</div>
`;
